graphql-dgs-codegen-gradle/src/main/kotlin/com/netflix/graphql/dgs/codegen/gradle/CodegenPlugin.kt

```diff
@@ -42,7 +44,9 @@ class CodegenPlugin : Plugin<Project> {
         generateJavaTaskProvider.configure { it.group = GRADLE_GROUP }
 
         val javaConvention = project.convention.getPlugin(JavaPluginConvention::class.java)
-        val sourceSets = javaConvention.sourceSets
+        val javaExtension = project.extensions.getByType(JavaPluginExtension::class.java)
+
+        val sourceSets = if (GradleVersion.current() >= GradleVersion.version("7.1")) javaExtension.sourceSets else javaConvention.sourceSets
```

Comment:
the conditional logic can be dropped once Gradle 7.x is not supported by this plugin

Reply:
@rpalcolea Can this be dropped yet? This is the last of my projects plugins using the deprecated convention and I was hoping to resolve this when I stumbled upon this PR and comment. Is there somewhere that documents the oldest version of Gradle that is intended to be supported?